### DISCUSSION TITLE
Conditional Puts/Upserts

### DISCUSSION THREAD

**Discussion by slloyd88** [PostgREST/postgrest#3444] [Q&A] [answered]
### Discussed in https://github.com/PostgREST/postgrest/discussions/3407

<div type='discussions-op-text'>

<sup>Originally posted by **slloyd88** April 12, 2024</sup>
I have a table called orders. The primary key is order_number. There is another numeric field called seq_num.  I am trying to find a way to do conditional upserts using PostgREST to only upsert if the new payload has a greater seq_num that the current record.

Any suggestions?</div>

There does not seem to be anyone in the discussion board that knows how to do this.  Can you please add it as a feature of PostgREST?

**Comment by laurenceisla** [comment, marked as answer]
I don't see a way to this directly with a `PUT` request. Perhaps it can be done by using `POST` and adding a `TRIGGER` that does the upsert as you mention, or by using a [function](https://postgrest.org/en/latest/references/api/functions.html) that does the upsert inside.

Let's keep the discussion here, I'm closing #3407 in favor of this one.

**Reply by steve-chavez** [reply]
Functions is definitely the way to go for now. I've noted this discussion on https://github.com/PostgREST/postgrest/issues/2602.
